features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/SampleAPI/SampleAPI.jsx

```diff
@@ -89,24 +89,22 @@ class SampleAPI extends Component {
         const { intl } = this.props;
         this.setState({ deploying: true });
         const promisedSampleAPI = this.createSampleAPI();
+        let api;
         const swaggerUpdatePromise = promisedSampleAPI.then((sampleAPI) => {
-            Alert.info(intl.formatMessage({
-                id: 'Apis.Listing.SampleAPI.SampleAPI.created',
-                defaultMessage: 'Sample PizzaShackAPI API created successfully',
-            }));
-            return sampleAPI.updateSwagger(getSampleSwagger('Unlimited'));
+            api = sampleAPI;
```

Comment:
I think this could lead to unexpected behaviors, Because we have declared the `api` variable in `handleDeploySample` method scope and assigning a value to it in the body of an asynchronous method,

We could have simply resolve `promisedSampleAPI` promise to `api` and use the api in `swaggerUpdatePromise ` then callback